chain-signatures/contract-eth/contracts/ChainSignatures.sol

```diff
@@ -44,14 +44,24 @@ contract ChainSignatures is Initializable {
 
     mapping(bytes32 => uint256) public depositToRefund;
 
+    uint32 public keyVersion;
+
     event SignatureRequested(bytes32 indexed requestId, address requester, uint256 epsilon, uint256 payloadHash, string path);
     event SignatureResponded(bytes32 indexed requestId, SignatureResponse response);
 
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor() initializer {}
+    constructor() {
+        _disableInitializers();
+    }
+
+    function initialize(PublicKey memory _publicKey, uint32 _keyVersion) public initializer {
+        publicKey = _publicKey;
+        keyVersion = _keyVersion;
+    }
```

Comment:
it's more gas efficient to use calldata since it is immutable and does not get allocated in heap
```suggestion
    function initialize(PublicKey calldata _publicKey, uint32 _keyVersion) public initializer {
        publicKey = _publicKey;
        keyVersion = _keyVersion;
    }
```